otwcode/otwarchive
# AO3-5901 Limit number of pseuds on profile and sidebar

# Pull Request Checklist


## Issue

https://otwarchive.atlassian.net/browse/AO3-5901

## Purpose

What does this PR do?

Limit the number of pseuds shown on the user's profile and on the user sidebar to the configured value in `ArchiveConfig.ITEMS_PER_PAGE`.

I have chosen to always include the default pseud as the first one, and then the rest of the pseuds alphabetically.

I've also repackaged the profile into a presenter, to keep the logic of arranging of data pulled from three different models (user, profile, preference) away from the view.

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended? (If you have access, please copy this into the JIRA ticket for them!)

Instructions are on the JIRA ticket. In order to test _before_ the nginx block is removed, one of the testy users might have to temporarily gain more than 25 pseuds...

## References

Are there any other relevant issues / PRs / mailing lists discussions related to this?

Yes, AO3-5889 - paginate pseuds, which I'm also working on.

## Credit

*What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?*

Enigel

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*

she/her
